### PR TITLE
ci: restructure reusable workflow into multiple calls

### PR DESCRIPTION
**Pull request by jedevc**
The core idea of this PR is to convert our giant `_dagger_call.yml` reusable workflow into a more concise, generic `call` action.

This has a couple of effects:
- It's a step towards being able to dogfood our own https://github.com/dagger/dagger-for-github action. We should be able to add a *ton* more functionality there someday, but we're a bit stuck until we can consume it.
- *Massively* simplifies the number of jobs for easier github viewing: compare
  > before:
  > image
  > after:
  > image
- We can easily combine multiple action calls in a single workflow - this reduces the total number of jobs, so that we need to spin up a fewer number of pods in total (hopefully reducing our total CI costs). This also ensures that these similar steps all occur on the same machine - which should improve our local caching performance.

### CODE REVIEW

**Comment by jedevc** [comment]
I think this *should* be ready for review now, but leaving in draft until I get back from holiday :palm_tree: so I can handle any after-merge issues.

File: .github/actions/call/action.yml
Comment (shykes) [inline .github/actions/call/action.yml:96]: Should I worry about this? :-P

Comment (jedevc) [reply .github/actions/call/action.yml:96]: Maybe - this is a only token to publish telemetry. It can't be made a secret, since these aren't consumable in PR workflows.

The "correct" mechansism for this is to eventually have OIDC flows supported in cloud. Depot has a fairly good explanation of how they do this: https://depot.dev/docs/cli/authentication#oidc-trust-relationships.

**Comment by gerhard** [comment]
I started looking into this last week, but didn't finish. This time I am saying a quick `👋 Hi` before I start digging into the review.

This PR will get merged quicker if we can address merge conflicts @jedevc

File: .github/actions/call/action.yml
Comment (gerhard) [reply .github/actions/call/action.yml:96]: @shykes No, I don't think so. This was here for at least 6 months. AFAIK, this is a write-only token. @marcosnils that has all the context.

File: .github/workflows/sdk-test.yml
Comment (gerhard) [inline .github/workflows/sdk-test.yml:1]: This is a great refactoring!

`SDK` name is the right one. Not as convinced about the filename: `sdk-test.yml`. This one seems more descriptive to me: `sdk.yml`

**Review by gerhard — APPROVED** [review]
I like this @jedevc! Left a few comments, nothing major.

I still need to check how these changes affect https://github.com/dagger/dagger/blob/2cbda9c7547809a35eb93ff7677e5bab4afd00c1/.github/workflows/trace-workflows.yml

I have a suspicion that we will know for sure only after we merge this.

Approving optimistically so that we don't hold this any longer than needed. It's been in flight long enough ✈️

**Comment by gerhard** [comment]
Once this is merged, DM me @jedevc so that I can double-check our Honeycomb integration.

File: .github/actions/call/action.yml
Comment (jedevc) [reply .github/actions/call/action.yml:90]: This is a pretty good point - the `--source` and `--host-docker-config` args are only valid inputs for our dagger/dagger module. For other modules, it's not valid.

We add these here so that most of our calls to this action (which is the dagger/dagger module) don't need to explicitly have this logic, but we can still call other modules with it if we want.

File: .github/workflows/_sdk_test.yml
Comment (jedevc) [reply .github/workflows/_sdk_test.yml:29]: :+1: will change!

File: .github/workflows/engine-and-cli-test.yml
Comment (jedevc) [reply .github/workflows/engine-and-cli-test.yml:80]: We should definitely be doing it here, so that we can detect vulnerable deps.

Not necessarily sure it has be done on publishing - but I do think it is useful, since then we have a public record of a scan done right after the publish, which could be useful to demonstrate that a release was good.

This was the behavior before fyi, happy to change it later!

File: .github/workflows/sdk-php-publish.yml
Comment (jedevc) [reply .github/workflows/sdk-php-publish.yml:9]: SGTM :tada:

Comment (gerhard) [reply .github/workflows/sdk-php-publish.yml]: My bad! We don't have the `dagger-v0-11-6-nvme` runner type. It's `dagger-v0-11-6`.

File: .github/workflows/engine-and-cli-test.yml
Comment (gerhard) [reply .github/workflows/engine-and-cli-test.yml]: OK, let's keep as is 👍

File: .github/workflows/sdk-php-publish.yml
Comment (jedevc) [reply .github/workflows/sdk-php-publish.yml]: :pray:

**Comment by gerhard** [comment]
Just checked all the changes again. When all workflows pass, we should and carry on with all the other changes 💪

**Comment by jedevc** [comment]
Hm, looks like downgrading to no nvme drives causes timeouts somehow.

**Comment by gerhard** [comment]
That's unfortunate but OK, let's stick with `dagger-v0-11-6-4c-nvme` for now. We can improve this part later.

File: .github/workflows/_dagger_call.yml
Comment (jedevc) [inline .github/workflows/_dagger_call.yml:84]: Hm, we need to make sure to keep the `concurrency` settings :eyes:

File: .github/actions/call/action.yml
Comment (marcosnils) [reply .github/actions/call/action.yml:96]: > @shykes No, I don't think so. This was here for at least 6 months. AFAIK, this is a telemetry write-only token (cannot write or read cache data). @marcosnils has all the context.

yes, this is correct. This is a "telemetry only" token, so it's ok to leave it where it is.

**Review by vito — APPROVED** [review]
:white_check_mark:  :chart_with_downwards_trend: :tada:

File: .github/workflows/sdk.yml
Comment (matipan) [reply .github/workflows/sdk.yml:58]: Taking care of dynamic sizing here: https://github.com/dagger/dagger.io/pull/3788. Once merged I will change the workflows here to use the correct sizes. For the SDKs we will use `4c` like we were doing previously, and for testdev we will continue with "32c" (NOTE: once the flakiness are solved I'm curious to see if 16c would be enough).

**Pull request merged**
